python/ray/dag/compiled_dag_node.py

```diff
@@ -407,49 +416,59 @@ def _read(self) -> bool:
         assert self._intermediate_buffer is None
         exit = False
         try:
-            input_data = self.input_reader.read()
-            self.set_intermediate_buffer(input_data)
+            future = self.input_reader.read()
```

Comment:
Hmm I think I am still not following why we need to expose futures to the channel interfaces? Why not keep the previous code but wrap `input_data` in a future?

Reply:
Done.